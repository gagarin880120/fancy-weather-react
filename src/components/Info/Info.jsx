import React from 'react';
import styles from './Info.module.css';
import AddressContainer from '../Address';
import CurrentDateContainer from '../CurrentDate';
import CurrentWeatherContainer from '../CurrentWeather';
import WeeklyWeather from '../WeeklyWeather/WeeklyWeather';

export default function Info() {
  return (
    <div className={styles.wrapper}>
      <AddressContainer />
      <CurrentDateContainer />
      <CurrentWeatherContainer />
      <WeeklyWeather />
    </div>
  );
}
