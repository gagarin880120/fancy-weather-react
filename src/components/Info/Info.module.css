.currentInfoMapContainer {
  display: flex;
  justify-content: space-between;
}

.currentInfoContainer {
  width: 60%;
}

@media screen and (max-width: 900px) {
  .currentInfoMapContainer {
    flex-direction: column;
  }

  .currentInfoContainer {
    width: 100%;
  }
}
