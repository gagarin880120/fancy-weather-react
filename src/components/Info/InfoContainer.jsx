import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Info from './Info';
import { selectPlace, selectCurrentWeather } from '../../redux/selectors';
import { getDefaultPlace } from '../../redux/actions';

export function InfoContainer({ place, currentWeather, getPlaceByIP }) {
  console.log('hi', place);

  useEffect(() => {
    getPlaceByIP();
  }, []);
  return (
    <Info
      place={place}
      currentWeather={currentWeather}
    />
  );
}

export const mapStateToProps = (state) => ({
  place: selectPlace(state),
  currentWeather: selectCurrentWeather(state),
});

export const mapDispatchToProps = (dispatch) => ({
  getPlaceByIP() {
    dispatch(getDefaultPlace());
  },
});

InfoContainer.propTypes = {
  place: PropTypes.string,
  currentWeather: PropTypes.shape({}),
  getPlaceByIP: PropTypes.func,
};

InfoContainer.defaultProps = {
  place: '',
  currentWeather: {},
  getPlaceByIP: () => {},
};

export default connect(mapStateToProps, mapDispatchToProps)(InfoContainer);
