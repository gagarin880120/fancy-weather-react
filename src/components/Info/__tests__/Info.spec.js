import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Info from '../Info';

jest.mock('mapbox-gl', () => ({
  Icons: null,
}));

describe('Info', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<Info />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
