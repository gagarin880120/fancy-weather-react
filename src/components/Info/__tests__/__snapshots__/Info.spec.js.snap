// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <div
    className="currentInfoMapContainer"
  >
    <div
      className="currentInfoContainer"
    >
      <Memo(Connect(AddressContainer)) />
      <Memo(Connect(CurrentDateContainer)) />
      <Memo(Connect(CurrentWeatherContainer)) />
    </div>
    <Memo(Connect(MapBoxContainer)) />
  </div>
  <Memo(Connect(WeeklyWeatherContainer)) />
</div>
`;
